[CI] test_arm result code is flaky

I saw this several times now, e.g, [here](https://github.com/ros-controls/ros2_control_ci/actions/runs/12227773179/job/34104976840)


```
    [test_position-8] SUCCEEDED result code
    [test_position-8] async_send_goal
Error: ROR] [test_position-8]: process has died [pid 40576, exit code 255, cmd '/home/runner/work/ros2_control_ci/ros2_control_ci/.work/target_ws/install/gz_ros2_control_tests/lib/gz_ros2_control_tests/test_position --ros-args'].
    FAIL
    
    ======================================================================
    FAIL: test_arm (gz_ros2_control_tests.TestFixture.test_arm))
    ----------------------------------------------------------------------
    Traceback (most recent call last):
      File "/home/runner/work/ros2_control_ci/ros2_control_ci/.work/target_ws/src/ros-controls/gz_ros2_control/gz_ros2_control_tests/tests/position_test.py", line 155, in test_arm
        launch_testing.asserts.assertExitCodes(proc_info, process=proc_action,
      File "/opt/ros/jazzy/lib/python3.12/site-packages/launch_testing/asserts/assert_exit_codes.py", line 62, in assertExitCodes
        assert info.returncode in allowable_exit_codes, 'Proc {} exited with code {}'.format(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    AssertionError: Proc test_position-8 exited with code 255
    
    ----------------------------------------------------------------------
    Ran 1 test in 9.157s
    
    FAILED (failures=1)
    [INFO] [python3-5]: sending signal 'SIGINT' to process[python3-5]
    [INFO] [robot_state_publisher-4]: sending signal 'SIGINT' to process[robot_state_publisher-4]
    [INFO] [parameter_bridge-2]: sending signal 'SIGINT' to process[parameter_bridge-2]
    [INFO] [gazebo-1]: sending signal 'SIGINT' to process[gazebo-1]
    [robot_state_publisher-4] [INFO] [1733712495.929158166] [rclcpp]: signal_handler(signum=2)
    [INFO] [python3-5]: process has finished cleanly [pid 40448]
    [parameter_bridge-2] [INFO] [1733712495.931029628] [rclcpp]: signal_handler(signum=2)
    [gazebo-1] Killed
Error: ROR] [gazebo-1]: process has died [pid 40443, exit code -2, cmd 'ruby $(which gz) sim --headless-rendering -s -r -v 4 empty.sdf --force-version 8'].
    [INFO] [robot_state_publisher-4]: process has finished cleanly [pid 40446]
    [INFO] [parameter_bridge-2]: process has finished cleanly [pid 40444]
```

--- a/gz_ros2_control_tests/src/test_position.cpp
+++ b/gz_ros2_control_tests/src/test_position.cpp
@@ -27,8 +27,8 @@ bool common_goal_accepted = false;
 rclcpp_action::ResultCode common_resultcode = rclcpp_action::ResultCode::UNKNOWN;
 int common_action_result_code = control_msgs::action::FollowJointTrajectory_Result::SUCCESSFUL;
 
-std::vector<double> derired_goals = {0, -1, 1, 0.0};
-unsigned int goal_reached = 0;
+std::vector<double> desired_goals = {0, -1, 1, 0.0};
+unsigned int ct_goals_reached = 0;
 
 void common_goal_response(
   rclcpp_action::ClientGoalHandle
@@ -74,16 +74,23 @@ void common_feedback(
   rclcpp_action::ClientGoalHandle<control_msgs::action::FollowJointTrajectory>::SharedPtr,
   const std::shared_ptr<const control_msgs::action::FollowJointTrajectory::Feedback> feedback)
 {
-  std::cout << "feedback->desired.positions :";
+  std::cout << "feedback->actual.positions :";
   for (auto & x : feedback->actual.positions) {
     std::cout << x << "\t";
   }
 
   std::cout << std::endl;
-  if (fabs(feedback->desired.positions[0] - derired_goals[goal_reached]) < 0.1) {
-    std::cout << "Goal " << derired_goals[goal_reached] << " reached" << '\n';
-    goal_reached++;
-    std::cout << "next goal " << goal_reached << '\n';
+  if (ct_goals_reached < desired_goals.size()) {
+    if (fabs(feedback->desired.positions[0] - desired_goals.at(ct_goals_reached)) < 0.1) {
+      std::cout << "Goal # " << ct_goals_reached << ": " << desired_goals.at(ct_goals_reached) <<
+        " reached" << std::endl;
+      ct_goals_reached++;
+      if (ct_goals_reached < desired_goals.size()) {
+        std::cout << "next goal # " << ct_goals_reached << ": " <<
+          desired_goals.at(ct_goals_reached) <<
+          std::endl;
+      }
+    }
   }
 }
 
@@ -124,25 +131,24 @@ int main(int argc, char * argv[])
   trajectory_msgs::msg::JointTrajectoryPoint point;
   point.time_from_start = rclcpp::Duration::from_seconds(0.0);  // start asap
   point.positions.resize(joint_names.size());
-
-  point.positions[0] = 0.0;
+  point.positions[0] = desired_goals[0];
 
   trajectory_msgs::msg::JointTrajectoryPoint point2;
   point2.time_from_start = rclcpp::Duration::from_seconds(1.0);
   point2.positions.resize(joint_names.size());
-  point2.positions[0] = -1.0;
+  point2.positions[0] = desired_goals[1];
 
   trajectory_msgs::msg::JointTrajectoryPoint point3;
   point3.time_from_start = rclcpp::Duration::from_seconds(2.0);
   point3.positions.resize(joint_names.size());
-  point3.positions[0] = 1.0;
+  point3.positions[0] = desired_goals[2];
 
   trajectory_msgs::msg::JointTrajectoryPoint point4;
   point4.time_from_start = rclcpp::Duration::from_seconds(3.0);
   point4.positions.resize(joint_names.size());
-  point4.positions[0] = 0.0;
+  point4.positions[0] = desired_goals[3];
 
-  // points.push_back(point);
+  points.push_back(point);
   points.push_back(point2);
   points.push_back(point3);
   points.push_back(point4);
@@ -157,6 +163,7 @@ int main(int argc, char * argv[])
   goal_msg.trajectory.joint_names = joint_names;
   goal_msg.trajectory.points = points;
 
+  std::cout << "async_send_goal" << std::endl;
   auto goal_handle_future = action_client->async_send_goal(goal_msg, opt);
 
   if (rclcpp::spin_until_future_complete(node, goal_handle_future) !=
@@ -194,10 +201,11 @@ int main(int argc, char * argv[])
   action_client.reset();
   node.reset();
 
-  std::cout << "async_send_goal" << std::endl;
+  std::cout << "call shutdown" << std::endl;
   rclcpp::shutdown();
 
-  if ((derired_goals.size() + 1) != goal_reached) {
+  if (desired_goals.size() != ct_goals_reached) {
+    std::cout << "Not all the goals were reached" << std::endl;
     return -1;
   }
 

